indico/web/forms/fields/principals.py

```diff
@@ -119,3 +124,22 @@ def process_formdata(self, valuelist):
         if valuelist:
             data = map(self._convert_principal, json.loads(valuelist[0]))
             self.data = None if not data else data[0]
+
+
+class PermissionsField(JSONField):
+    widget = JinjaWidget('forms/permissions_widget.html', single_kwargs=True, acl=True)
+
+    def __init__(self, *args, **kwargs):
+        super(PermissionsField, self).__init__(*args, **kwargs)
+        self.ip_networks = map(serialize_ip_network_group, IPNetworkGroup.query.filter_by(hidden=False))
+
+    @property
+    def event(self):
+        return self.get_form().event
+
+    @property
+    def roles(self):
+        return [serialize_role(role) for role in self.get_form().event.roles]
+
+    def _value(self):
+        return json.dumps(self.data, separators=(',', ':'), ensure_ascii=False).encode('utf-8') if self.data else '[]'
```

Comment:
Why `separators=(',', ':'), ensure_ascii=False`?

Reply:
I was having troubles with characters like 'ö' but I will change it.

Reply:
Maybe a bytestring was passed somewhere?